pkg/cloudprovider/fake/capacity.go

```diff
@@ -42,11 +44,11 @@ func (c *Capacity) Create(ctx context.Context, constraints *cloudprovider.Constr
 				Taints:     constraints.Taints,
 			},
 			Status: v1.NodeStatus{
+				// Right size the instance
 				Allocatable: v1.ResourceList{
-					// The (fake) mythical mega-machine
-					v1.ResourcePods:   resource.MustParse("1000000"),
-					v1.ResourceCPU:    resource.MustParse("1000000"),
-					v1.ResourceMemory: resource.MustParse("1000000Gi"),
+					v1.ResourcePods:   resource.MustParse(fmt.Sprint(len(constraints.Pods) + 2)),
```

Comment:
I'm imagining the `+2` is for `kube-system` pods. Since this is a fake cloud provider, is this necessary? Could we have more than 2 default daemonsets?

Reply:
It's a bit awkward. Even worse since no one actually cares about this pod number. I wish I had the ability to derive the number of daemonsets from the overhead, but it's currently collapsed into a single value. 

I'd be willing to set this back to 1000000.